src/vmm/src/builder.rs

```diff
@@ -471,20 +458,17 @@ fn choose_payload(vm_resources: &VmResources) -> Result<Payload, StartMicrovmErr
         #[cfg(all(target_os = "linux", target_arch = "x86_64", not(feature = "tee")))]
         return Ok(Payload::KernelMmap);
 
-        #[cfg(all(target_arch = "aarch64", not(feature = "efi")))]
+        #[cfg(target_arch = "aarch64")]
         return Ok(Payload::KernelCopy);
     } else if let Some(external_kernel) = vm_resources.external_kernel() {
         Ok(Payload::ExternalKernel(external_kernel.clone()))
+    } else if cfg!(feature = "efi") {
+        Ok(Payload::Efi)
     } else {
         Err(StartMicrovmError::MissingKernelConfig)
     }
 }
 
-#[cfg(feature = "efi")]
-fn choose_payload(_vm_resources: &VmResources) -> Result<Payload, StartMicrovmError> {
-    Ok(Payload::Efi)
-}
-
```

Comment:
This should probably go in the previous commit, in my opinion

Reply:
Yup, fixing it.